mysite/api/views.py

```diff
@@ -15,3 +15,13 @@ class QuestionViewSet(viewsets.ModelViewSet):
 class ChoiceViewSet(viewsets.ModelViewSet):
     serializer_class = ChoiceSerializer
     queryset = Choice.objects.all()
+
+
+class UserViewSet(viewsets.ModelViewSet):
+    serializer_class = UserRetrieveSerializer
+    queryset = User.objects.all()
+    serializer_action_classes = {
+        'list': UserListSerializer,
+        'create': UserCreateSerializer,
+        'update': UserUpdateSerializer
+    }
```

Comment:
Is this being used? I don't think DRF supports it

Reply:
It is being used exclusively in this new UserViewSet. It seems to be working fine. I'll write tests on it to check it is being properly supported.
I used the following article as reference.
[https://medium.com/aubergine-solutions/decide-serializer-class-dynamically-based-on-viewset-actions-in-django-rest-framework-drf-fb6bb1246af2](url)

Reply:
It works but if you handle return in the serializer in the get_serializer_class method of the viewset I think. By it's own it won't make anything

Reply:
Oh, now I see. Yes, even the very article I use as an argument explains what you both are saying. I did not override the get_serializer_class method. I'll get on it ASAP.